### ISSUE TITLE
offline expire date

### ISSUE THREAD

**Issue by shinriyo** [Esri/arcgis-maps-sdk-swift-samples#261]
Does it have offline expire date setting?

I want to set more than 1 month.

**Comment by yo1995** [comment]
Yes, the new API does support the [expiration on the mobile map package](https://developers.arcgis.com/swift/api-reference/documentation/arcgis/mobilemappackage/expiration). We don't have an example written for it yet, but you can take a look at the old one [here](https://github.com/Esri/arcgis-runtime-samples-ios/tree/main/arcgis-ios-sdk-samples/Maps/Honor%20mobile%20map%20package%20expiration%20date). I'll take note and prioritize this sample.

**Comment by shinriyo** [comment]
Thank you for replying
What is the default expiration?

**Comment by yo1995** [comment]
If you author the mobile map package in ArcGIS Pro, you'll be prompt to select an expiration date when the "Enable Map Expiration" is checked. [doc](https://pro.arcgis.com/en/pro-app/latest/tool-reference/data-management/create-mobile-map-package.htm)

![image](https://github.com/Esri/arcgis-maps-sdk-swift-samples/assets/9660181/5f956a19-41e4-471b-a1f4-b1fe041620de)

I think I misread your question though - Unfortunately, mobile map packages created with the [OfflineMapTask](https://developers.arcgis.com/swift/api-reference/documentation/arcgis/offlinemaptask) do not support expiration. I believe it's due to some licensing intricacy. 😥 So you'll need to author the mmpk on Pro.

**Comment by yo1995** [comment]
> If the expiration date can be turned off I would like to turn it off.

This need to be done when the mmpk is authored, either by disabling the expiration, or specify `allowExpiredAccess` (allow to open). If it is specified an `preventExpiredAccess` (Do not allow to open) option, then it cannot be accessed after expired.

**Comment by shinriyo** [comment]
Will offline data be deleted automatically? Or will it remain forever unless I delete it?

**Comment by yo1995** [comment]
> Will offline data be deleted automatically? Or will it remain forever unless I delete it?

It depends on where you specify the [offline download directory](https://github.com/Esri/arcgis-maps-sdk-swift-samples/blob/main/Shared/Samples/Generate%20offline%20map/GenerateOfflineMapView.swift#L198) like in this sample.

If you specify a system temporary or cache directory, then it will be purged by the system automatically.

If you specify the app's document directory or other custom  folder you created, then you need to manage the files. Think the offline map data as files, and you can manage them like normal files.

- https://developer.apple.com/library/archive/documentation/FileManagement/Conceptual/FileSystemProgrammingGuide/FileSystemOverview/FileSystemOverview.html
- https://developer.apple.com/documentation/foundation/url/3988456-documentsdirectory

**Comment by shinriyo** [comment]
![Screen Shot 2023-09-05 at 16 27 47](https://github.com/Esri/arcgis-maps-sdk-swift-samples/assets/396927/b5b02827-9c0d-4b56-9fab-5a6a943f7d7e)


I ran the sample that you describe `[offline download directory](https://github.com/Esri/arcgis-maps-sdk-swift-samples/blob/v.next/Shared/Samples/Generate%20offline%20map/GenerateOfflineMapView.swift#L206) l`

But I can't understand what it is.

**Comment by yo1995** [comment]
> I ran the sample that you describe [offline download directory](https://github.com/Esri/arcgis-maps-sdk-swift-samples/blob/v.next/Shared/Samples/Generate%20offline%20map/GenerateOfflineMapView.swift#L206)
> But I can't understand what it is.

Please read the [instructions](https://github.com/Esri/arcgis-maps-sdk-swift-samples#configuring-api-keys) in the main README to set up an API key. When exporting an offline map using an Esri basemap service, you'll need an API key to get the tiles.

This sample uses the default generate offline map parameters which set the `continuesOnErrors` property to `true`. Therefore, the task proceeded even with the error, and the output in your screenshot is incomplete.

**Comment by shinriyo** [comment]
<img width="659" alt="スクリーンショット 2023-09-06 16 18 20" src="https://github.com/Esri/arcgis-maps-sdk-swift-samples/assets/396927/0093e80e-1913-466c-a7bb-b856aa337641">
I can't create key. Should I pay for the key?
So, I can't try just sample.

**Comment by yo1995** [comment]
> I can't create key. Should I pay for the key? So, I can't try just sample.

Please see the [limitations](https://developers.arcgis.com/documentation/mapping-apis-and-services/security/api-keys/#limitations) of different account types. Did you follow the instructions in the root README and sign up for a developer account from the [developer website](https://developers.arcgis.com/dashboard/)?
